_posts/2017-01-16-Jak-psat-php-Uloziste-kodu+databaze+frameworky.md

```diff
@@ -0,0 +1,46 @@
+---
+layout: post
+title: Jak psát PHP? (3/3) - Úložiště kódu, databáze a frameworky
+---
+
+V poslední části přehledu o nástrojích vhodných pro jazyk PHP se podíváme nejen na github + gitlab, řekneme si něco málo o databázích a předestřu základy frameworků. Tady můžete najít [druhý](http://jakpsatphp.cz/Jak-psat-php-Zavislosti+verzovaci-systemy+debugging/) nebo [první](http://jakpsatphp.cz/Jak-psat-php-Zdroje-znalosti+vyvojove-prostredi/) díl série. 
+
+# Úložiště kódu
+Když nebudete pracovat sami nebo se budete chtít pochlubit novým projektem,
+ 
+Všechny úložiště poskytují základní funkce jako jsou větvení a klonování kódu, Pull Requesty, Code Review, Issues nebo Markdown Support. 
+
+Nejpoužívanějším úložištěm je [github.com](https://github.com). Je úzce provázaný s gitem a díky němu je možné snadno nahrávat, stahovat, opravovat i větvit cizí i vlastní kód. Jedinnou nevýhodou je možnost soukromých repozitářů pouze v placené verzi. 
+
+[Gitlab.com](https://about.gitlab.com/) je mladší služba, která proti githubu nabízí i soukromé repozitáře a jako jeden z mála úložišť je opensourcovaný a můžete si ho tedy zdarma nasadit i do vlastního řešení. Oproti githubu ale nenabízí sledování jednotlivých uživatelů. 
+
+[Bitbucket](https://bitbucket.org/) není pro projekty tak rozšířený jako předchozí varianty, jako jediný ale nabízí podporu [verzovacího systému Mercurial](https://www.mercurial-scm.org/). Navíc velmi ochotně spolupracuje s dalšími aplikacemi od firmy [Atlassian](https://www.atlassian.com/) jako je [Confluence](https://www.atlassian.com/software/confluence) nebo [Jira](https://www.atlassian.com/software/jira). 
+
+Mezi dalšími chci zmínit [CodePlex](https://www.codeplex.com/) Mezi služby, které nechi doporučovat patří [SourceForge](https://sourceforge.net/), který měl veliké problémy s přibalováním nevyžádaných probramů do všech projektů. Používal se i [Google Code](https://code.google.com/archive/), který ale v roce 2016 ukončil svou činnost a je dostupný jen ve formě archívu. 
```

Comment:
Tady mi pak chybí nějaké srování nebo doporučení. Proč a kdy mám po kterém sáhnout.

Co používáš ty a proč.

Zase bych nepropagoval mrtvé projekty (poslední odstavec), to akorát zmate a vyvolá otázky.

*Mezi služby, které nechi doporučovat patří* - krásný oxymoron. Nechceš doporučovat, ale zmiňuješ je, tedy svým způsobem doporučuješ. Tuhle větu a vše až do konce bych vyhodil.